### PR TITLE
VZ-4951 Introduce Fluentd log parsing for Grafana

### PR DESCRIPTION
**Pull request by davisgomes**
# Description

Creates a Fluentd log parsing section for Grafana component and implements a subsequent acceptance test.

Fixes VZ-4951

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-logging.yaml [hunk 1]
```diff
@@ -453,6 +453,41 @@ data:
...
+      @type parser
+      @id grafana
+      key_name log
+      reserve_data true
+      emit_invalid_record_to_error true
+      <parse>
+        # Grafana has one formats for log records
+        @type multi_format
+        <pattern>
+          format /^t=(?<logtime>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2})\+\d{4} lvl=(?<@level>\S+) msg="(?<@message>.*?)".*?$/
```
Comment (bobdonat) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-logging.yaml:468]: why is @level and @message needed here

Comment (davisgomes) [reply platform-operator/helm_config/charts/verrazzano/templates/verrazzano-logging.yaml:468]: I have updated the Ruby parsing to get rid of this. Testing now. Will update soon.

File: tests/e2e/logging/system/system_logging_test.go [hunk 2]
```diff
@@ -278,6 +279,16 @@ func validateCertManagerLogs() bool {
 		noExceptions)
 }
 
+func validateGrafanaLogs() bool {
+	return validateElasticsearchRecords(
+		basicElasticsearchRecordValidator,
+		systemIndex,
+		"app",
```
Comment (bobdonat) [inline tests/e2e/logging/system/system_logging_test.go:286]: is "app" correct here?  I this being used for other components - kubernetes.labels.app.keyword

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-logging.yaml [hunk 3]
```diff
@@ -453,6 +453,47 @@ data:
...
+  
+    # filter to parse Grafana container log files
+    <filter kubernetes.**vmi-system-grafana**verrazzano-system_grafana**>
+      @type parser
+      @id grafana
+      key_name log
+      reserve_data true
+      emit_invalid_record_to_error true
+      <parse>
+        # Grafana has one formats for log records
```
Comment (bobdonat) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-logging.yaml:465]: comment says Grafana has one format but multiple formats are parsed.

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-logging.yaml [hunk 4]
```diff
@@ -453,6 +453,47 @@ data:
...
+    # filter to apply a record transformer Grafana container log files in JSON format
+    <filter kubernetes.**vmi-system-grafana**verrazzano-system_grafana**>
+      @type record_transformer
+      @id grafana-json
+      key_name log
+      reserve_data true
+      emit_invalid_record_to_error true
+      enable_ruby
+      <record>
+        message ${record["@message"] ? record["@message"] : record["message"]}
```
[the remark this answers is not in the document]
Comment (davisgomes) [reply platform-operator/helm_config/charts/verrazzano/templates/verrazzano-logging.yaml:492]: You're right. I should have a fail case.

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-logging.yaml (hunk 3 above)
Comment (davisgomes) [reply platform-operator/helm_config/charts/verrazzano/templates/verrazzano-logging.yaml]: Fixed

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**